### PR TITLE
Fix large angle changes in lattice primitive generator

### PR DESCRIPTION
**Pull request by mattbooker**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #2808 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | N/A |

---

## Description of contribution in a few bullet points

* Removed any large angular changes in the lattice motion primitives
* Added more tests for the lattice generator

## Description of documentation updates required from your changes

None needed.

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_smac_planner/lattice_primitives/trajectory_generator.py [hunk 1]
```diff
@@ -566,3 +569,39 @@ def generate_trajectory(
         trajectory_path = self._create_path(trajectory_params, primitive_resolution)
 
         return Trajectory(trajectory_path, trajectory_params)
+
+if __name__ == "__main__":
+    test = TrajectoryGenerator({"turning_radius":0.5})
```
Comment (SteveMacenski) [inline nav2_smac_planner/lattice_primitives/trajectory_generator.py:574]: Should this be here?

Comment (mattbooker) [reply nav2_smac_planner/lattice_primitives/trajectory_generator.py]: removed it in latest. Forgot to remove my testing code...

**Comment by SteveMacenski** [comment]
Can you regenerate the existing primitive files so that they have this fix?

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
https://github.com/ros-planning/navigation2/blob/main/nav2_smac_planner/test/default_model.json#L1794

it also looks like the default 0.4m one in `test/` needs to be regenerated too, not just the 0.5m/1.0m ones in the generator

File: nav2_smac_planner/lattice_primitives/helper.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 def normalize_angle(angle):
     """
-    Normalize the angle to between [-pi, pi).
+    Normalize the angle to between (-pi, pi].
```
Comment (SteveMacenski) [inline nav2_smac_planner/lattice_primitives/helper.py:20]: I'll need to test this locally too with a few of these files. I thought we needed 0-2PI not -PI to PI. I don't see how it was working before looking at the parsers https://github.com/ros-planning/navigation2/blob/main/nav2_smac_planner/src/node_lattice.cpp but clearly it was so I need to understand what's going on there

File: nav2_smac_planner/lattice_primitives/helper.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 def normalize_angle(angle):
     """
-    Normalize the angle to between [-pi, pi).
+    Normalize the angle to between [0, pi).
```
Comment (SteveMacenski) [inline nav2_smac_planner/lattice_primitives/helper.py:20]: 0-2PI

**Pull request merged**